chain/src/types.rs

```diff
@@ -232,15 +233,27 @@ pub struct OutputRoots {
 }
 
 impl OutputRoots {
+	/// The root of our output PMMR. The rules here are block height specific.
+	/// We use the merged root here for header version 3 and later.
+	pub fn root(&self, header: &BlockHeader) -> Result<Hash, Error> {
+		if !consensus::valid_header_version(header.height, header.version) {
```

Comment:
i don't see why all these utility functions need to error on incorrect header version. ideally there is only one place where correct header version is checked, and other functions just assume it correct.

but in this case, I don't even see why the root function depends on the header. It should only depend on header version, in which case it can not error either.

Reply:
Fair point - I can clean this up.